Session.php

```diff
@@ -165,6 +165,21 @@ public function isValid()
         return true;
     }
 
+    /**
+     * Callback service's logout()
+     * @return void
+     */
+    public function logout() {
+        $cookie = $this->getCookie();
+        if (!$cookie) return;
+        /** @var \helper_plugin_oauth $hlp */
+        $hlp = plugin_load('helper', 'oauth');
+        $service = $hlp->loadService($cookie['servicename']);
+        if ($service === null) return;
+        $service->initOAuthService($cookie['storageId']);
+        $service->logout();
+    }
```

Comment:
this is wrong here. it has nothing to do with the session and cookie management.